openml/extensions/sklearn/extension.py

```diff
@@ -782,15 +785,18 @@ def _get_external_version_string(
     def _check_multiple_occurence_of_component_in_flow(
         self,
         model: Any,
-        sub_components: Dict[str, Any],
+        sub_components: Dict[str, Union[OpenMLFlow, str, None]],
     ) -> None:
-        to_visit_stack = []  # type: List[OpenMLFlow]
+        to_visit_stack = []  # type: List[Union[OpenMLFlow, str, None]]
         to_visit_stack.extend(sub_components.values())
         known_sub_components = set()  # type: Set[str]
+
         while len(to_visit_stack) > 0:
             visitee = to_visit_stack.pop()
-            if isinstance(visitee, str):
+            if isinstance(visitee, str):  # 'drop', 'passthrough' can be passed as estimators
                 known_sub_components.add(visitee)
+            elif visitee is None:  # a None step can be included in a Pipeline
```

Comment:
Can we maybe simplify the code and only support `passthrough` instead of `None`? It appears to me that `passthrough` can be used in the same places and is much clearer to handle. Maybe raise an error message if a user puts in a `None` step?

Reply:
On checking further, [FeatureUnion supports _drop_ and _None_](https://github.com/openml/openml-python/blob/cbf0d6a749a4f23245f15668a548f235c61b82f6/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L678) while [Pipeline supports _passthrough_ and _None_](https://github.com/openml/openml-python/blob/cbf0d6a749a4f23245f15668a548f235c61b82f6/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L1804).

Should we still _not_ support None and raise an Exception along the lines of: 
`None not allowed, please use 'drop' or 'passthrough'`?

However, when creating the sklearn Pipeline object, using _drop_ flags an error.
Whereas for FeatureUnion, using _passthrough_ is not flagged until runtime, i.e., calling _fit_.

Not sure what would be the best design here, especially wrt the Exception message, if we do not allow None.